docs/src/examples/Cloudy_example.md

```diff
@@ -0,0 +1,518 @@
+using CalibrateEmulateSample.Utilities
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+using EnsembleKalmanProcesses.DataContainers
+```
+
+#### Load the calibration results
+
+We will train the emulator on the input-output pairs we obtained during the
+calibration. They are stored within the `EnsembleKalmanProcess` object
+(`ekiobj`), which is loaded here together with the other information that was
+saved in the calibration step.
+
+```julia
+ekiobj = load(data_save_file)["eki"]
+priors = load(data_save_file)["priors"]
+truth_sample_mean = load(data_save_file)["truth_sample_mean"]
+truth_sample = load(data_save_file)["truth_sample"]
+# True parameters:
+# - ϕ: in constrained space
+# - θ: in unconstrained space
+ϕ_true = load(data_save_file)["truth_input_constrained"]
+θ_true = transform_constrained_to_unconstrained(priors, ϕ_true)
+```
+
+The user can choose one or both of two emulators: a Gaussian Process (GP) emulator with `GaussianProcesses.jl` interface (`gp-gpjl`) and a scalar Random Feature
+(RF) interface (`rf-scalar`). See
+[here](https://clima.github.io/CalibrateEmulateSample.jl/previews/PR257/examples/emulators/regression_2d_2d)
+for a complete overview of the available emulators. 
+
+```julia
+cases = [
+    "rf-scalar",
+    "gp-gpjl"  # Veeeery slow predictions
+]
+```
+
+We first define some settings for the two emulators, e.g., the prediction type
+for the GP emulator, or the number of features and hyperparameter optimizer
+options for the RF emulator. The docs for GPs and RFs (TODO: add links!) explain the different options in more detail and provide some useful heuristics for how to customize the settings depending on the problem at hand.
+
+```julia
+# These settings are the same for all Gaussian Process cases
+pred_type = YType() # we want to predict data
+
+# These settings are the same for all Random Feature cases
+n_features = 400 
+nugget = 1e-8
+optimizer_options = Dict(
+    "verbose" => true,
+    "scheduler" => DataMisfitController(terminate_at = 100.0),
+    "cov_sample_multiplier" => 1.0,
+    "n_iteration" => 20,
+)
+```
+
+Emulation is performed through the construction of an `Emulator` object from the
+following components:
+* a wrapper for the machine learning tool to be used as emulator
+* the input-output pairs on which the emulator will be trained
+* optional arguments specifying data processing and dimensionality reduction
+  functionality 
+
+For `gp-gpjl`, this looks as follows:
+```julia
+gppackage = GPJL()
+# Kernel is the sum of a squared exponential (SE), Matérn 5/2, and white noise
+gp_kernel = SE(1.0, 1.0) + Mat52Ard(zeros(3), 0.0) + Noise(log(2.0))
+
+# Wrapper for GP
+gaussian_process = GaussianProcess(
+    gppackage;
+    kernel = gp_kernel,
+    prediction_type = pred_type,
+    noise_learn = false,
+)
+
+# Use all input-output pairs available from the EKI iterations
+input_output_pairs = get_training_points(ekiobj,
```

Comment:
These are the same for both options. as will the emulator (see my comment on the example) so really the GPJL and SRFI ones should only differ in the creation of the ML tool specifically that gets passed into the emulator

Reply:
Yep, I restructured the example such that the `input_output_pairs` only get defined once (see comment above)